pcstac/pcstac/main.py

```diff
@@ -75,6 +76,15 @@
 
 add_timeout(app, app_settings.request_timeout)
 
+
+@app.middleware("http")
+async def _request_middleware(
+    request: Request, call_next: Callable[[Request], Awaitable[Response]]
+) -> Response:
+    """Add a trace to all requests."""
+    return await trace_request(ServiceName.STAC, request, call_next)
```

Comment:
Confirmed that this approach does work though, I see the expected custom dimensions in the App Insights table.